### PR TITLE
[Improvement] state/elasticache-redis: add MultiAZ option

### PR DESCRIPTION
**Pull request by vicinus**
Adds the option to run elasticache redis instanzes in different AZs for ha.

### CODE REVIEW

File: state/elasticache-redis.yaml [hunk 1]
```diff
@@ -124,6 +125,11 @@ Parameters:
     Default: 1
     MinValue: 0
     MaxValue: 5
+  MultiAZ:
```
Comment (andreaswittig) [inline state/elasticache-redis.yaml:128]: I would argue, that MultiAZ should be enabled by default when NumReplicas > 0? Can you think of any reason to disable MultiAZ?

**Comment by andreaswittig** [comment]
Thanks a lot for your contribution, @vicinus.

File: state/elasticache-redis.yaml (hunk 1 above)
Comment (michaelwittig) [reply state/elasticache-redis.yaml:128]: The docs mention this:

> Multi-AZ is only supported on Redis clusters that have more than one node in each shard.

**Comment by vicinus** [comment]
I can think of no reason to not enable Multi-AZ. I think if the cluster has more than one node, then the only reason, to not enable it  is, if your setup uses only one AZ. As the templates do not support a one AZ setup, it should be fine to enable Multi-AZ automatically if the cluster has more than one node.
If you are fine with that I will change the request and remove the Multi-AZ parameter and activate the Multi-AZ option with a condition automatically if the number of nodes are more than one.

**Comment by michaelwittig** [comment]
@vicinus that sounds great!

**Comment by vicinus** [comment]
@andreaswittig Sorry, my inexperience with github shows again. I thought I could merge my two commits prior to updating the pull request...
So is there a way to save this pull request or should we close it and I create a new clean pull request with only one commit?

**Comment by andreaswittig** [comment]
@vicinus No worries. Feel free to push as many commits as needed. We will stash the commits when merging this PR. So at the end there will be a single commit.

**Comment by vicinus** [comment]
Thanks, perfect. The result after applying both commits should be only the line:
`MultiAZEnabled: !If [HasAutomaticFailoverEnabled, true, false]`
I think it is correct to reuse the condition HasAutomaticFailoverEnabled but if you prefer a dedicated condition I can change that.

**Comment by michaelwittig** [comment]
@vicinus sometimes things are easier than expected :) One last question: Are you running this new template already?

**Comment by vicinus** [comment]
I did do some tests: I created a redis cluster with the original template and one Replica and checked that the MultiAZ option was deactivated. I then updated the cluster with the updated template and checked that the MultiAZ option was activated. Afterwards I reduced the number of Replicas to 0 and checked that the MultiAZ was deactivated. But I did not change any of our productive clusters yet, because I thought I wait till the change is accepted.

**Comment by michaelwittig** [comment]
Sounds great. We also run our automated test suite so we should be fine. Thank you!

**Review by michaelwittig — APPROVED** [review]

**Comment by michaelwittig** [comment]
@vicinus [v13.1.0](https://github.com/widdix/aws-cf-templates/releases/tag/v13.1.0) contains your change

**Pull request merged**